### PR TITLE
Fixes from 0907 xpd test

### PR DESCRIPTION
**Pull request by CJ-Wright**

### CODE REVIEW

**Comment by ordirules** [comment]
can you write some quick lines with the inputs and what each version gives (if you have time)? Just would save me a little bit of time. Thanks!

**Comment by CJ-Wright** [comment]
Sure.

File: shed/event_streams.py [hunk 1]
```diff
@@ -1018,6 +1050,22 @@ def update(self, x, who=None):
         elif len(L) > self.maxsize:
             return self.condition.wait()
 
+    def _truncate_update(self, x, who=None):
+        L = self.buffers[self.children.index(who)]
+        L.append(x)
+        if len(L) == 1 and all(self.buffers):
+            for i in range(len(self.buffers)):
+                # If the docs don't match, do nothing
```
Comment (ordirules) [inline shed/event_streams.py]: "if the docs don't match, throw away" maybe?

**Comment by CJ-Wright** [comment]
Yes the zip logic can be rather difficult to work with. Currently nothing really supports multiple descriptors which is a job for after the cycle starts in my mind. I will try to modify the test to see if it can work in the event model.

**Pull request merged**